src/nvim/CMakeLists.txt

```diff
@@ -199,7 +200,7 @@ add_custom_command(OUTPUT ${GENERATED_UNICODE_TABLES}
 )
 
 add_custom_command(OUTPUT ${GENERATED_API_DISPATCH} ${API_METADATA}
-    COMMAND ${LUA_PRG} ${DISPATCH_GENERATOR} ${PROJECT_SOURCE_DIR}/scripts ${API_HEADERS} ${GENERATED_API_DISPATCH} ${API_METADATA}
+  COMMAND ${LUA_PRG} ${DISPATCH_GENERATOR} ${API_SOURCE_DIR} ${API_HEADERS} ${GENERATED_API_DISPATCH} ${API_METADATA}
```

Comment:
Seems unnecessary to have a variable for every subdir, no? if we should follow the existing pattern
`set(DEPRECATED_DISPATCH_FILE ${PROJECT_SOURCE_DIR}/src/nvim/api/deprecated_dispatch.lua` but when thinking on it why not have `set(NVIM_SOURCE_DIR ${PROJECT_SOURCE_DIR}/src/nvim)` and then write `${NVIM_SOURCE_DIR}/api/deprecated_dispatch.lua`, `${NVIM_SOURCE_DIR}/ex_cmds.lua` etc in the build rules?